### PR TITLE
Cps/empire app to co sim app

### PR DESCRIPTION
**Pull request by BasselSaridar**
closes #5955

### CODE REVIEW

File: applications/CoSimulationApplication/tests/test_CoSimulationApplication.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
     from test_coupling_operations import TestScalingOperation
     from test_flower_coupling import TestFLOWerCoupling
     from test_sdof_solver import TestSdofSolver
+    from test_sdof_static_solver import TestSdofStaticSolver
```
Comment (philbucher) [inline applications/CoSimulationApplication/tests/test_CoSimulationApplication.py:24]: you should also add the fsi-sdof here
check the co_sim_test_factory

File: applications/CoSimulationApplication/tests/test_sdof_static_solver.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        system.Initialize()
+
+        system.SolveSolutionStep()
+        system.OutputSolutionStep()
+
+        results_obtained = np.loadtxt('fsi_sdof_static/results_final_sdof.dat', skiprows=1)
+        results_reference = np.loadtxt('reference_files/ref_sdof_static_final_displacement.dat', skiprows=1)
+        self.assertEqual(results_reference.all(), results_obtained.all())
+
+    def test_wrapper_variables_check(self):
```
Comment (philbucher) [inline applications/CoSimulationApplication/tests/test_sdof_static_solver.py]: please remove this test, this is being checked in the baseclass already

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+class PotentialFlowWrapper(kratos_base_wrapper.KratosBaseWrapper):
+    def _CreateAnalysisStage(self):
+        return PotentialFlowAnalysis(self.model, self.project_parameters)
+
+    def AdvanceInTime(self, current_time):
+        self.time = 1.0
+        return self.time
+    def Predict(self):
+        return
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py]:
```suggestion
        pass
```

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+def Create(settings, solver_name):
+    return PotentialFlowWrapper(settings, solver_name)
+
+class PotentialFlowWrapper(kratos_base_wrapper.KratosBaseWrapper):
+    def _CreateAnalysisStage(self):
+        return PotentialFlowAnalysis(self.model, self.project_parameters)
+
+    def AdvanceInTime(self, current_time):
+        self.time = 1.0
+        return self.time
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py]:
```suggestion
        return 0.0
```

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+def Create(settings, solver_name):
+    return PotentialFlowWrapper(settings, solver_name)
+
+class PotentialFlowWrapper(kratos_base_wrapper.KratosBaseWrapper):
+    def _CreateAnalysisStage(self):
+        return PotentialFlowAnalysis(self.model, self.project_parameters)
+
+    def AdvanceInTime(self, current_time):
+        self.time = 1.0
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py]:
```suggestion

```

File: applications/CoSimulationApplication/tests/fsi_mok/ProjectParametersCFD.json [hunk 6]
```diff
@@ -0,0 +1,303 @@
+{
```
Comment (philbucher) [inline applications/CoSimulationApplication/tests/fsi_mok/ProjectParametersCFD.json]: this is a temp file, should not be committed

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            if sub_project_parameters[i]["python_module"].GetString() == "compute_lift_process":
+                self.lift_process = ComputeLiftProcess(self.model, sub_project_parameters[i]["Parameters"])
+
+    def SolveSolutionStep(self):
+        self.wake_process.ExecuteInitialize()
+        self._analysis_stage._GetSolver().fluid_solver.solver.Clear()
+        self._analysis_stage._GetSolver().fluid_solver.solver.InitializeSolutionStep()
+        super(PotentialFlowWrapper, self).SolveSolutionStep()
+        self.model["FluidModelPart"].CloneTimeStep(self.time)
+        self.model["FluidModelPart"].ProcessInfo[KratosMultiphysics.STEP] += 1
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py]:
```suggestion
```

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+                self.conversion_process = ComputeForcesOnNodesProcess(self.model, sub_project_parameters[i]["Parameters"])
+            if sub_project_parameters[i]["python_module"].GetString() == "compute_lift_process":
+                self.lift_process = ComputeLiftProcess(self.model, sub_project_parameters[i]["Parameters"])
+
+    def SolveSolutionStep(self):
+        self.wake_process.ExecuteInitialize()
+        self._analysis_stage._GetSolver().fluid_solver.solver.Clear()
+        self._analysis_stage._GetSolver().fluid_solver.solver.InitializeSolutionStep()
+        super(PotentialFlowWrapper, self).SolveSolutionStep()
+        self.model["FluidModelPart"].CloneTimeStep(self.time)
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py]:
```suggestion
```

File: applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                    raise Exception("potential flow requires specification of a process for the wake (currently specifically using 'define_wake_process_2d')")
+
+            if sub_project_parameters[i]["python_module"].GetString() == "compute_forces_on_nodes_process":
+                self.conversion_process = ComputeForcesOnNodesProcess(self.model, sub_project_parameters[i]["Parameters"])
+            if sub_project_parameters[i]["python_module"].GetString() == "compute_lift_process":
+                self.lift_process = ComputeLiftProcess(self.model, sub_project_parameters[i]["Parameters"])
+
+    def SolveSolutionStep(self):
+        self.wake_process.ExecuteInitialize()
+        self._analysis_stage._GetSolver().fluid_solver.solver.Clear()
```
Comment (philbucher) [inline applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/potential_flow_wrapper.py]: please add a comment why this is necessary

**Review by philbucher — COMMENTED** [review]
almost ready :+1:

**Comment by philbucher** [comment]
@BasselSaridar can you please come to see me?

**Comment by philbucher** [comment]
~~~py
class SdofSolverWrapper:
    def __init__(self, settings, solver_name):
        CoSimulationSolverWrapper.__init__(self,settings, solver_name)
        self._sdof_solver = CreateSdofSolver()


    @classmethod
    def CreateSdofSolver(self):
        return SDoFStatic(...)

class SdofStaticSolverWrapper(SdofSolverWrapper):
    """ This class implements a wrapper for an SDof solver to be used in CoSimulation
    """
    def __init__(self, settings, solver_name):
        CoSimulationSolverWrapper.__init__(self,settings, solver_name)

        input_file_name = self.settings["solver_wrapper_settings"]["input_file"].GetString()

        self.mp = self.model.CreateModelPart("Sdof_Static")
        self.mp.ProcessInfo[KM.DOMAIN_SIZE] = 1

    @classmethod
    def CreateSdofSolver(cls):
        return SDoFStaticSolver(input_file_name)
~~~

**Comment by inigolcanalejo** [comment]
Hi @philbucher, I went through your comments and made the suggested changes. If there is something else, let me know.

Also when running the tests I get a nice Segmentation fault (core dumped), Ill open another PR to explain the issue.

**Comment by philbucher** [comment]
nice, I will take a look!



> Also when running the tests I get a nice Segmentation fault (core dumped), Ill open another PR to explain the issue.

:heart: :heart: :heart:

**Comment by inigolcanalejo** [comment]
Although the tests pass here, they don't do so in my machine, neither in FullDebug nor in Release, do the tests pass in your machine @philbucher?

**Comment by inigolcanalejo** [comment]
I tried recompiling from scratch but it also does not work. This is my version of gcc in case it matters:
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609

**Review by philbucher — APPROVED** [review]
ok I think it is time to merge this :)

It works on my machine, I did some minor modifications

**Pull request merged**